openfisca_survey_manager/scenarios/abstract_scenario.py

```diff
@@ -0,0 +1,682 @@
+        for simulation_name, simulation in self.simulations.items():
+            print(f"simulation : {simulation_name}")  # noqa analysis:ignore
+            simulation.print_memory_usage()
+
+    def neutralize_variables(self, tax_benefit_system):
+        """Neutralizes input variables not in input dataframe and keep some crucial variables.
+
+        Args:
+          tax_benefit_system: The TaxBenefitSystem variables belongs to
+
+        """
+        for variable_name, variable in tax_benefit_system.variables.items():
+            if variable.formulas:
+                continue
+            if self.used_as_input_variables and (variable_name in self.used_as_input_variables):
+                continue
+            if self.non_neutralizable_variables and (variable_name in self.non_neutralizable_variables):
+                continue
+            if self.weight_variable_by_entity and variable_name in list(self.weight_variable_by_entity.values()):
+                continue
+
+            tax_benefit_system.neutralize_variable(variable_name)
+
+    def restore_simulations(self, directory, **kwargs):
+        """Restores SurveyScenario's simulations.
+
+        Args:
+          directory: Directory to restore simulations from
+          kwargs: Restoration options
+
+        """
+        assert os.path.exists(directory), "Cannot restore simulations from non existent directory"
+        use_sub_directories = True if len(self.tax_benefit_systems) >= 2 else False
+
+        self.simulations = dict()
+        if use_sub_directories:
+            for simulation_name, tax_benefit_system in self.tax_benefit_systems.items():
+                simulation = restore_simulation(
+                    os.path.join(directory, simulation_name),
+                    tax_benefit_system,
+                    **kwargs)
+                simulation.id_variable_by_entity_key = self.id_variable_by_entity_key
+                self.simulations[simulation_name] = simulation
+        else:
+            simulation = restore_simulation(directory, list(self.tax_benefit_systems.values())[0], **kwargs)
+            simulation.id_variable_by_entity_key = self.id_variable_by_entity_key
+            self.simulations["unique_simulation"] = simulation
+
+    def set_input_data_frame(self, input_data_frame):
+        """Set the input dataframe.
+
+        Args:
+          input_data_frame (pd.DataFrame): Input data frame
+
+        """
+        self.input_data_frame = input_data_frame
+
+    def set_tax_benefit_systems(self, tax_benefit_systems: dict, reference: str = None):
+        """Set the tax and benefit systems of the scenario.
+
+        Args:
+          tax_benefit_systems:
+        """
+        for tax_benefit_system in tax_benefit_systems.values():
+            assert tax_benefit_system is not None
+            if self.cache_blacklist is not None:
+                tax_benefit_system.cache_blacklist = self.cache_blacklist
+        self.reference_tax_benefit_system = (
```

Comment:
@benjello you define this `reference_tax_benefit_system` but it's not used neither in the AbstractSurveyScenario nor in the tests. So could you explain when and why should we use it ? Thanks !

Reply:
You are right, I should have cleaned this after removing its occurences eveywhere else